examples/qwen_3_vl/qwen_3_vl.zig

```diff
@@ -0,0 +1,1211 @@
+                    deepstack_features_list[count] = zml.call(self.deepstack_patch_mergers[count], .forward, .{ hidden_states, true });
+                    count += 1;
+                }
+            }
+        }
+        hidden_states = zml.call(self.patch_merger, .forward, .{ hidden_states, false });
+        return .{ hidden_states, deepstack_features_list };
+    }
+
+    // Positional embedding interpolation (representation of the image in a grid determined by the number of position embeddings 48 x 48)
+    pub fn fastPosEmbedInterpolate(self: VisionTransformer, grid: []const u32) [1]Tensor {
+        // Calculate the number of grid points per side (sqrt of the number of position embeddings)
+        const num_grid_per_side = std.math.pow(f32, @as(f32, @floatFromInt(self.num_position_embeddings)), 0.5);
+
+        const m_size = self.spatial_merge_size;
+        const embedding_dim = self.hidden_size;
+
+        var outputs = [1]Tensor{undefined};
+
+        // Retrieve the dims in the image grid
+        const t = grid[0];
+        const h = grid[1];
+        const w = grid[2];
+        const tensor_filled_1_h = zml.Tensor.constant(.{h}, zml.Data.init(.f32, 1));
+        const tensor_filled_1_w = zml.Tensor.constant(.{w}, zml.Data.init(.f32, 1));
+
+        // Build the indices for the height and width by linearly spacing the grid points
+        const h_idxs = zml.Tensor.linspace(.{ .start = 0, .end = num_grid_per_side - 1, .steps = h }, .f32);
+        const w_idxs = zml.Tensor.linspace(.{ .start = 0, .end = num_grid_per_side - 1, .steps = w }, .f32);
+        const h_floor = h_idxs.floor();
+        const w_floor = w_idxs.floor();
+
+        // Build the min and max clamp for the height and width
+        const min_clamp_h = zml.Tensor.constant(.{h}, zml.Data.init(.f32, 0));
+        const max_clamp_h = zml.Tensor.constant(.{h}, zml.Data.init(.f32, num_grid_per_side - 1));
+        const min_clamp_w = zml.Tensor.constant(.{w}, zml.Data.init(.f32, 0));
+        const max_clamp_w = zml.Tensor.constant(.{w}, zml.Data.init(.f32, num_grid_per_side - 1));
+
+        // Build the ceil and floor for the height and width
+        const h_ceil = h_floor.add(tensor_filled_1_h).clamp(min_clamp_h, max_clamp_h);
+        const w_ceil = w_floor.add(tensor_filled_1_w).clamp(min_clamp_w, max_clamp_w);
+
+        // Build the difference between the indices and the floor for the height and width -> delta with the grid points
+        const dh = h_idxs.sub(h_floor);
+        const dw = w_idxs.sub(w_floor);
+
+        // Build the meshgrid for the height and width
+        const tensor_filled_1_h_v = zml.Tensor.constant(.{ h, w }, zml.Data.init(.f32, 1));
+        const d_tensors_meshgrid = [2]Tensor{ dh, dw };
+        const floor_tensors_meshgrid = [2]Tensor{ h_floor, w_floor };
+        const ceil_tensors_meshgrid = [2]Tensor{ h_ceil, w_ceil };
+        const dhw_grid = zml.torch.meshgrid(2, d_tensors_meshgrid, .ij);
```

Comment:
call `cartesianProduct` directly. Meshgrid is only useful if you need the .xy layout for some reason.